src/AutoRest.CSharp/Common/Output/Models/Types/SerializableObjectType.cs

```diff
@@ -58,13 +70,33 @@ protected SerializableObjectType(string defaultNamespace, SourceInputModel? sour
             return _xmlSerialization;
         }
 
+        private BicepObjectSerialization? EnsureBicepSerialization()
+        {
+            if (_bicepSerializationInitialized)
+                return _bicepSerialization;
+
+            _bicepSerializationInitialized = true;
+            _bicepSerialization = BuildBicepSerialization();
+            return _bicepSerialization;
+        }
+
         protected abstract JsonObjectSerialization? BuildJsonSerialization();
         protected abstract XmlObjectSerialization? BuildXmlSerialization();
 
+        protected BicepObjectSerialization? BuildBicepSerialization()
+        {
+            // if this.Usages does not contain Output bit, then return null
+            // alternate - is one of ancestors resource data or contained on a resource data
+            var usage = GetUsage();
+
```

Comment:
This should be updated to only build bicep for models that are either ResourceData (or derive from ResourceData) or are properties on a ResourceData (or a model that derives from ResourceData).